src/test/java/com/yscope/logging/log4j1/TestRollingFileLogAppender.java

```diff
@@ -292,34 +436,39 @@ private void validateNumSyncAndCloseEvents (RollingFileTestAppender appender, in
   }
 
   /**
-   * Creates and initializes a RollingFileTestAppender with the given
-   * rollover sizes
-   * @param compressedRolloverSize
-   * @param uncompressedRolloverSize
+   * Creates and initializes a RollingFileTestAppender for the tests. Note that this method doesn't
+   * call {@code activateOptions} on the appender.
+   * @param disableSoftTimeout
+   * @param disableHardTimeout
    * @return The created appender
    */
-  private RollingFileTestAppender createTestAppender (int compressedRolloverSize,
-                                                      int uncompressedRolloverSize,
-                                                      boolean disableSoftTimeout,
-                                                      boolean disableHardTimeout)
-  {
+  private RollingFileTestAppender createTestAppender (boolean disableSoftTimeout,
+                                                      boolean disableHardTimeout) {
     RollingFileTestAppender appender = new RollingFileTestAppender();
-    // Parameters from AbstractClpIrBufferedRollingFileAppender
-    appender.setRolloverCompressedSizeThreshold(compressedRolloverSize);
-    appender.setRolloverUncompressedSizeThreshold(uncompressedRolloverSize);
+
+    // Set static settings
     appender.setOutputDir(outputDir);
     appender.setBaseName("test-file");
     appender.setCloseFrameOnFlush(true);
-    // Parameters from AbstractBufferedRollingFileAppender
-    appender.setCloseFileOnShutdown(true);
     appender.setLayout(patternLayout);
     appender.setTimeoutCheckPeriod(timeoutCheckPeriod);
-    String disabledTimeoutCsv = "ERROR=" + Integer.MAX_VALUE + ",INFO=" + Integer.MAX_VALUE;
-    String timeoutCsv = "ERROR=" + flushErrorLevelTimeout + ",INFO=" + flushInfoLevelTimeout;
+    appender.setRolloverCompressedSizeThreshold(Integer.MAX_VALUE);
+    appender.setRolloverUncompressedSizeThreshold(Integer.MAX_VALUE);
+
+    // Set timeouts
+    String disabledTimeoutCsv =
+        "FATAL=" + Integer.MAX_VALUE + ",ERROR=" + Integer.MAX_VALUE + ", INFO="
+            + Integer.MAX_VALUE;
+    String timeoutCsv =
+        "FATAL=" + flushFatalLevelTimeout + ",ERROR=" + flushErrorLevelTimeout + ",INFO="
+            + flushInfoLevelTimeout;
     appender.setFlushHardTimeoutsInMinutes(disableHardTimeout ? disabledTimeoutCsv : timeoutCsv);
     appender.setFlushSoftTimeoutsInSeconds(disableSoftTimeout ? disabledTimeoutCsv : timeoutCsv);
 
-    appender.activateOptions();
     return appender;
   }
+
+  private void triggerContextSwitch () {
```

Comment:
Improve the documentation of why we need to trigger context switch, i.e unit test runs in single thread.